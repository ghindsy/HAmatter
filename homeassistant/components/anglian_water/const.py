"""Anglian Water consts."""

from logging import Logger, getLogger

LOGGER: Logger = getLogger(__package__)

NAME = "Anglian Water"
DOMAIN = "anglian_water"

CONF_DEVICE_ID = "device_id"
