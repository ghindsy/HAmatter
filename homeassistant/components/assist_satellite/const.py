"""Constants for assist satellite."""

DOMAIN = "assist_satellite"
