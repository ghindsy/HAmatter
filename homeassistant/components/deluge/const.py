"""Constants for the Deluge integration."""

import enum
import logging
from typing import Final

CONF_WEB_PORT = "web_port"
CURRENT_STATUS = "current_status"
DEFAULT_NAME = "Deluge"
DEFAULT_RPC_PORT = 58846
DEFAULT_WEB_PORT = 8112
DOMAIN: Final = "deluge"
DOWNLOAD_SPEED = "download_speed"

LOGGER = logging.getLogger(__package__)
PROTOCOL_TRAFFIC_DOWNLOAD_SPEED = "protocol_traffic_download_speed"
PROTOCOL_TRAFFIC_UPLOAD_SPEED = "protocol_traffic_upload_speed"
UPLOAD_SPEED = "upload_speed"


class DelugeGetSessionStatusKeys(enum.Enum):
    """Enum representing the keys that get passed into the Deluge RPC `core.get_session_status` xml rpc method.

    you can call `core.get_session_status` with no keys (so an empty list in deluge-client.DelugeRPCClient.call)
    to get the full list of possible keys, but it seems to basically be a all of the session statistics
    listed on this page: https://www.rasterbar.com/products/libtorrent/manual-ref.html#session-statistics
    and a few others

    there is also a list of deprecated keys that deluge will translate for you and issue a warning in the log:
    https://github.com/deluge-torrent/deluge/blob/7f3f7f69ee78610e95bea07d99f699e9310c4e08/deluge/core/core.py#L58

    """

    DHT_DOWNLOAD_RATE = "dht_download_rate"
    DHT_UPLOAD_RATE = "dht_upload_rate"
    DOWNLOAD_RATE = "download_rate"
    UPLOAD_RATE = "upload_rate"
