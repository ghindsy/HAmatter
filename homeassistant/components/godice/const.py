"""Constants for the godice integration."""

from datetime import timedelta

DOMAIN = "godice"
SCAN_INTERVAL = timedelta(seconds=60)

MANUFACTURER = "Particula"
MODEL = "GoDice"

CONF_SHELL = "shell"

DICE_MIN_NUMBER = 1
DICE_MAX_NUMBER = 20
