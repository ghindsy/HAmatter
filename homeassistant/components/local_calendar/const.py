"""Constants for the Local Calendar integration."""

DOMAIN = "local_calendar"

CONF_CALENDAR_NAME = "calendar_name"
CONF_ICS_FILE = "ics_file"
CONF_IMPORT_ICS = "import_ics"
CONF_STORAGE_KEY = "storage_key"
STORAGE_PATH = ".storage/local_calendar.{key}.ics"
