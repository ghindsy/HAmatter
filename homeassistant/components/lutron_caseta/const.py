"""Lutron Caseta constants."""

from typing import Final

DOMAIN = "lutron_caseta"

CONF_KEYFILE = "keyfile"
CONF_CERTFILE = "certfile"
CONF_CA_CERTS = "ca_certs"

STEP_IMPORT_FAILED = "import_failed"
ERROR_CANNOT_CONNECT = "cannot_connect"
ABORT_REASON_CANNOT_CONNECT = "cannot_connect"

LUTRON_CASETA_BUTTON_EVENT = "lutron_caseta_button_event"

BRIDGE_DEVICE_ID = "1"

DEVICE_TYPE_WHITE_TUNE = "WhiteTune"
DEVICE_TYPE_SPECTRUM_TUNE = "SpectrumTune"

MANUFACTURER = "Lutron Electronics Co., Inc"

ATTR_SERIAL: Final = "serial"
ATTR_TYPE: Final = "type"
ATTR_BUTTON_TYPE: Final = "button_type"
ATTR_LEAP_BUTTON_NUMBER: Final = "leap_button_number"
ATTR_BUTTON_NUMBER: Final = "button_number"  # LIP button number
ATTR_DEVICE_NAME: Final = "device_name"
ATTR_AREA_NAME: Final = "area_name"
ATTR_ACTION: Final = "action"

ACTION_PRESS = "press"
ACTION_RELEASE = "release"

CONF_SUBTYPE = "subtype"

BRIDGE_TIMEOUT = 35

UNASSIGNED_AREA = "Unassigned"

CONFIG_URL = "https://device-login.lutron.com"
