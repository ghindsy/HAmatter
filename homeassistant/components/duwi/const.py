"""Constants for the Duwi integration.

This module contains constants that are used throughout the Duwi Smart Hub integration,
including configuration keys, default values, and URLs for API endpoints.
"""

from homeassistant.const import Platform

# Unique domain identifier for the Duwi Smart Hub integration
DOMAIN = "duwi"

# Manufacturer of the product for identification purposes
MANUFACTURER = "Duwi"

# Version of the App for the Duwi Smart Hub integration, used for tracking and compatibility.
APP_VERSION = "0.1.1"

# Home Assistant client version
CLIENT_VERSION = "0.1.1"

# Model identification of the Home Assistant client, typically used for logging or diagnostics.
CLIENT_MODEL = "homeassistant"

# List of platforms that support config entry
SUPPORTED_PLATFORMS = [Platform.SWITCH]

SENSOR_TYPE = "7"
