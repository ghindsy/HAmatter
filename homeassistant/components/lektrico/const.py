"""Constants for the Lektrico Charging Station integration."""

from logging import Logger, getLogger

from homeassistant.const import Platform

# Integration domain
DOMAIN = "lektrico"

# Logger
LOGGER: Logger = getLogger(__package__)

# List the platforms that charger supports.
CHARGERS_PLATFORMS = [Platform.SENSOR]

# List the platforms that load balancer device supports.
LB_DEVICES_PLATFORMS = [Platform.SENSOR]
