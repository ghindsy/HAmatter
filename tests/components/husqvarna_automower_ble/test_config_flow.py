"""Test the Husqvarna Bluetooth config flow."""

from unittest.mock import Mock

from bleak import BleakError
import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant import config_entries
from homeassistant.components.husqvarna_automower_ble.const import DOMAIN
from homeassistant.config_entries import SOURCE_USER
from homeassistant.core import HomeAssistant

from . import (
    AUTOMOWER_MISSING_SERVICE_SERVICE_INFO,
    AUTOMOWER_SERVICE_INFO,
    AUTOMOWER_UNNAMED_SERVICE_INFO,
    AUTOMOWER_UNSUPPORTED_GROUP_SERVICE_INFO,
)

from tests.components.bluetooth import inject_bluetooth_service_info

pytestmark = pytest.mark.usefixtures("mock_setup_entry")


async def test_user_selection(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test we can select a device."""

    inject_bluetooth_service_info(hass, AUTOMOWER_SERVICE_INFO)
    inject_bluetooth_service_info(hass, AUTOMOWER_UNNAMED_SERVICE_INFO)
    await hass.async_block_till_done(wait_background_tasks=True)

    result = await hass.config_entries.flow.async_init(
        DOMAIN, context={"source": SOURCE_USER}
    )
    assert result == snapshot

    result = await hass.config_entries.flow.async_configure(
        result["flow_id"],
        user_input={"address": "00000000-0000-0000-0000-000000000001"},
    )
    assert result == snapshot

    result = await hass.config_entries.flow.async_configure(
        result["flow_id"],
        user_input={},
    )


async def test_no_devices(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test missing device."""

    inject_bluetooth_service_info(hass, AUTOMOWER_MISSING_SERVICE_SERVICE_INFO)
    inject_bluetooth_service_info(hass, AUTOMOWER_UNSUPPORTED_GROUP_SERVICE_INFO)
    await hass.async_block_till_done(wait_background_tasks=True)

    result = await hass.config_entries.flow.async_init(
        DOMAIN, context={"source": SOURCE_USER}
    )
    assert result == snapshot


async def test_bluetooth(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test bluetooth device discovery."""

    inject_bluetooth_service_info(hass, AUTOMOWER_SERVICE_INFO)
    await hass.async_block_till_done(wait_background_tasks=True)

    result = hass.config_entries.flow.async_progress_by_handler(DOMAIN)[0]
    assert result == snapshot

    result = await hass.config_entries.flow.async_configure(
        result["flow_id"],
        user_input={},
    )


async def test_bluetooth_invalid(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test bluetooth device discovery with invalid data."""

    inject_bluetooth_service_info(hass, AUTOMOWER_UNSUPPORTED_GROUP_SERVICE_INFO)
    await hass.async_block_till_done(wait_background_tasks=True)

    result = await hass.config_entries.flow.async_init(
        DOMAIN,
        context={"source": config_entries.SOURCE_BLUETOOTH},
        data=AUTOMOWER_UNSUPPORTED_GROUP_SERVICE_INFO,
    )
    assert result == snapshot


async def test_failed_connect(
    hass: HomeAssistant,
    mock_client: Mock,
    snapshot: SnapshotAssertion,
) -> None:
    """Test we can select a device."""

    inject_bluetooth_service_info(hass, AUTOMOWER_SERVICE_INFO)
    inject_bluetooth_service_info(hass, AUTOMOWER_UNNAMED_SERVICE_INFO)
    await hass.async_block_till_done(wait_background_tasks=True)

    mock_client.connect.side_effect = False

    result = await hass.config_entries.flow.async_init(
        DOMAIN, context={"source": config_entries.SOURCE_USER}
    )
    assert result == snapshot

    result = await hass.config_entries.flow.async_configure(
        result["flow_id"],
        user_input={"address": "00000000-0000-0000-0000-000000000001"},
    )
    assert result == snapshot

    result = await hass.config_entries.flow.async_configure(
        result["flow_id"],
        user_input={},
    )


async def test_exception_connect(
    hass: HomeAssistant,
    mock_client: Mock,
    snapshot: SnapshotAssertion,
) -> None:
    """Test we can select a device."""

    inject_bluetooth_service_info(hass, AUTOMOWER_SERVICE_INFO)
    inject_bluetooth_service_info(hass, AUTOMOWER_UNNAMED_SERVICE_INFO)
    await hass.async_block_till_done(wait_background_tasks=True)

    async def _probe_gatts(self, device):
        """Mock BleakClient.probe_gatts."""
        raise BleakError

    mock_client.probe_gatts = _probe_gatts


async def test_failed_is_connected(
    hass: HomeAssistant,
    mock_client: Mock,
    snapshot: SnapshotAssertion,
) -> None:
    """Test we can select a device."""

    inject_bluetooth_service_info(hass, AUTOMOWER_SERVICE_INFO)
    inject_bluetooth_service_info(hass, AUTOMOWER_UNNAMED_SERVICE_INFO)
    await hass.async_block_till_done(wait_background_tasks=True)

    mock_client.is_connected.side_effect = False
