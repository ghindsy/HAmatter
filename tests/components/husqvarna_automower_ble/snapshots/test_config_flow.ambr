# serializer version: 1
# name: test_bluetooth
  FlowResultSnapshot({
    'context': dict({
      'confirm_only': True,
      'source': 'bluetooth',
      'title_placeholders': dict({
        'name': 'Husqvarna Automower',
      }),
      'unique_id': '00000000-0000-0000-0000-000000000003',
    }),
    'flow_id': <ANY>,
    'handler': 'husqvarna_automower_ble',
    'step_id': 'confirm',
  })
# ---
# name: test_bluetooth_invalid
  FlowResultSnapshot({
    'description_placeholders': None,
    'flow_id': <ANY>,
    'handler': 'husqvarna_automower_ble',
    'reason': 'no_devices_found',
    'type': <FlowResultType.ABORT: 'abort'>,
  })
# ---
# name: test_failed_connect
  FlowResultSnapshot({
    'data_schema': list([
      dict({
        'name': 'address',
        'required': True,
        'type': 'string',
      }),
    ]),
    'description_placeholders': None,
    'errors': None,
    'flow_id': <ANY>,
    'handler': 'husqvarna_automower_ble',
    'last_step': None,
    'preview': None,
    'step_id': 'user',
    'type': <FlowResultType.FORM: 'form'>,
  })
# ---
# name: test_failed_connect.1
  FlowResultSnapshot({
    'data_schema': None,
    'description_placeholders': dict({
      'name': 'Husqvarna Automower',
    }),
    'errors': None,
    'flow_id': <ANY>,
    'handler': 'husqvarna_automower_ble',
    'last_step': None,
    'preview': None,
    'step_id': 'confirm',
    'type': <FlowResultType.FORM: 'form'>,
  })
# ---
# name: test_failed_is_connected
  FlowResultSnapshot({
    'data_schema': list([
      dict({
        'name': 'address',
        'required': True,
        'type': 'string',
      }),
    ]),
    'description_placeholders': None,
    'errors': None,
    'flow_id': <ANY>,
    'handler': 'husqvarna_automower_ble',
    'last_step': None,
    'preview': None,
    'step_id': 'user',
    'type': <FlowResultType.FORM: 'form'>,
  })
# ---
# name: test_failed_is_connected.1
  FlowResultSnapshot({
    'description_placeholders': dict({
      'error': '',
    }),
    'flow_id': <ANY>,
    'handler': 'husqvarna_automower_ble',
    'reason': 'cannot_connect',
    'type': <FlowResultType.ABORT: 'abort'>,
  })
# ---
# name: test_no_devices
  FlowResultSnapshot({
    'data_schema': list([
      dict({
        'name': 'address',
        'required': True,
        'type': 'string',
      }),
    ]),
    'description_placeholders': None,
    'errors': None,
    'flow_id': <ANY>,
    'handler': 'husqvarna_automower_ble',
    'last_step': None,
    'preview': None,
    'step_id': 'user',
    'type': <FlowResultType.FORM: 'form'>,
  })
# ---
# name: test_user_selection
  FlowResultSnapshot({
    'data_schema': list([
      dict({
        'name': 'address',
        'required': True,
        'type': 'string',
      }),
    ]),
    'description_placeholders': None,
    'errors': None,
    'flow_id': <ANY>,
    'handler': 'husqvarna_automower_ble',
    'last_step': None,
    'preview': None,
    'step_id': 'user',
    'type': <FlowResultType.FORM: 'form'>,
  })
# ---
# name: test_user_selection.1
  FlowResultSnapshot({
    'data_schema': None,
    'description_placeholders': dict({
      'name': 'Husqvarna Automower',
    }),
    'errors': None,
    'flow_id': <ANY>,
    'handler': 'husqvarna_automower_ble',
    'last_step': None,
    'preview': None,
    'step_id': 'confirm',
    'type': <FlowResultType.FORM: 'form'>,
  })
# ---
